### PR TITLE
Service API Backend

### PR DESCRIPTION
**Pull request by wburklund**

### CODE REVIEW

File: api/QCVOC.Api/Service/Controller/ServiceController.cs [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        /// <param name="service">The updated Service.</param>
+        /// <returns>See attributes.</returns>
+        /// <response code="200">The Service was updated successfully.</response>
+        /// <response code="400">The specified Service was invalid.</response>
+        /// <response code="401">Unauthorized.</response>
+        /// <response code="404">A Service matching the specified id could not be found.</response>
+        /// <response code="409">A Service with the same name already exists.</response>
+        /// <response code="500">The server encountered an error while processing the request.</response>
+        [HttpPut("{id}")]
+        [ProducesResponseType(typeof(Service), 200)]
```
Comment (jpdillingham) [inline api/QCVOC.Api/Service/Controller/ServiceController.cs]: Let's restrict this to just admins and supervisors: `[Authorize(Roles = nameof(Role.Administrator) + "," + nameof(Role.Supervisor))]`

File: api/QCVOC.Api/Service/Controller/ServiceController.cs [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        /// </summary>
+        /// <param name="service">The Service to create.</param>
+        /// <returns>See attributes.</returns>
+        /// <response code="201">The Service was created successfully.</response>
+        /// <response code="400">The specified Service was invalid.</response>
+        /// <response code="401">Unauthorized.</response>
+        /// <response code="409">A Service with the same name already exists.</response>
+        /// <response code="500">The server encountered an error while processing the request.</response>
+        [HttpPost("")]
+        [Authorize]
```
Comment (jpdillingham) [inline api/QCVOC.Api/Service/Controller/ServiceController.cs]: `[Authorize(Roles = nameof(Role.Administrator) + "," + nameof(Role.Supervisor))]`

File: api/QCVOC.Api/Service/Controller/ServiceController.cs [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+        /// <response code="401">Unauthorized.</response>
+        /// <response code="500">The server encountered an error while processing the request.</response>
+        [HttpGet("")]
+        [Authorize]
+        [ProducesResponseType(typeof(IEnumerable<Service>), 200)]
+        [ProducesResponseType(401)]
+        [ProducesResponseType(typeof(Exception), 500)]
+        public IActionResult GetAll([FromQuery]ServiceFilters filters)
+        {
+            return Ok(ServiceRepository.GetAll(filters));
```
Comment (jpdillingham) [inline api/QCVOC.Api/Service/Controller/ServiceController.cs]: Missing the singular `Get()`

File: api/QCVOC.Api/Service/Controller/ServiceController.cs (hunk 1 above)
Comment (wburklund) [reply api/QCVOC.Api/Service/Controller/ServiceController.cs]: Mea culpa

File: api/QCVOC.Api/Service/Controller/ServiceController.cs (hunk 3 above)
Comment (wburklund) [reply api/QCVOC.Api/Service/Controller/ServiceController.cs]: I wasn't sure this was necessary, but it's in now

**Pull request merged**